cg/apps/lims/limsjson.py

```diff
@@ -1,52 +1,70 @@
 from cg.apps.lims.orderform import CASE_PROJECT_TYPES, expand_case
-from cg.constants import Pipeline
+from cg.constants import DataDelivery, Pipeline
 from cg.exc import OrderFormError
+from cg.meta.orders import OrderType
 from cg.meta.orders.status import StatusHandler
 
-OPTIONAL_KEYS = (
-    "container_name",
-    "quantity",
-    "volume",
-    "concentration",
-    "status",
-    "comment",
-    "capture_kit",
-    "mother",
-    "father",
-)
+ACCEPTED_DATA_ANALYSES = [{str(Pipeline.MIP_DNA)}, {str(Pipeline.FLUFFY)}, {str(Pipeline.BALSAMIC)}]
 
 
 def get_project_type(samples: [dict]) -> str:
     """Determine the project type."""
 
     data_analyses = set(sample.get("data_analysis", "mip-dna").lower() for sample in samples)
 
-    if len(data_analyses) != 1:
+    if len(data_analyses) > 1:
         raise OrderFormError(f"mixed 'Data Analysis' types: {', '.join(data_analyses)}")
 
-    if data_analyses == {"mip-dna"}:
-        project_type = "mip-dna"
-    elif data_analyses == {"fluffy"}:
-        project_type = "rml"
-    elif data_analyses == {"balsamic"}:
-        project_type = "balsamic"
-    else:
-        raise OrderFormError(f"Unsupported json orderform: {data_analyses}")
+    if data_analyses in ACCEPTED_DATA_ANALYSES:
+        return data_analyses.pop()
+
+    raise OrderFormError(f"Unsupported order_data orderform: {data_analyses}")
+
+
+def get_data_delivery(samples: [dict], project_type: OrderType) -> str:
+    """Determine the order_data delivery type."""
+
+    NO_VALUE = "no_value"
+    data_deliveries = set(sample.get("data_delivery", NO_VALUE).lower() for sample in samples)
+
+    if len(data_deliveries) > 1:
+        raise OrderFormError(f"mixed 'Data Delivery' types: {', '.join(data_deliveries)}")
+
+    data_delivery = data_deliveries.pop()
+
+    if data_delivery == NO_VALUE:
+        if project_type == OrderType.METAGENOME:
+            return str(DataDelivery.FASTQ)
+        elif project_type == OrderType.FASTQ:
```

Comment:
Like code factor says these could (should?) be `if` instead of `elif` since there are return statements for everyone

Reply:
fixed